mod_app/admin/link_admin.py

```diff
@@ -148,10 +170,43 @@ class OtherLinkInline(PreviewMixin, admin.TabularInline):
     ]
 
 
-@admin.register(Video)
-class VideoAdmin(PreviewMixin, admin.ModelAdmin):
-    search_fields = ["description", "url"]
-    list_display = ["description", "film", "url", "preview"]
+# Registering models based off the Filelink abstract class
+for model in [Video]:
+    register_custom_admin(
+        model=model,
+        mixins=(PreviewMixin),
+        search_fields=["description", "url"],
+        list_display=["description", "film", "url", "preview"],
+        autocomplete_fields=["archive"],
+    )
+
+for model in [CardImage]:
+    register_custom_admin(
+        model=model,
+        mixins=(PreviewMixin, s3BrowserButtonMixin),
+        search_fields=["description", "url"],
+        list_display=["description", "film", "url", "preview"],
+        exclude=["archive"],
+    )
+
+for model in [
+    Script,
+    PressBook,
+    Programme,
+    Publicity,
+    Still,
+    Postcard,
+    Poster,
+    Drawing,
+    PublicVisualInfluence,
+]:
+    register_custom_admin(
+        model=model,
+        mixins=(PreviewMixin, s3BrowserButtonMixin),
+        search_fields=["description", "url"],
+        list_display=["description", "film", "file", "url", "preview"],
+        autocomplete_fields=["archive"],
```

Comment:
this actually needs to be added to all the inlines 😅 and is also a tuple format again

might be worth using another loop for defining these inlines since most of them are identical